library/ssl_tls.c

```diff
@@ -4387,16 +4381,75 @@ static int ssl_another_record_in_datagram( mbedtls_ssl_context *ssl )
 
 static int ssl_load_buffered_message( mbedtls_ssl_context *ssl )
 {
-    /* No buffering support so far. */
-    ((void) ssl );
-    return( -1 );
+    mbedtls_ssl_handshake_params * const hs = ssl->handshake;
+    int ret = 0;
+
+    MBEDTLS_SSL_DEBUG_MSG( 2, ( "=> ssl_load_buffered_messsage" ) );
+
+    if( hs == NULL )
```

Comment:
I moved the check above the debug message, but why do you prefer `if( ssl->handshake == NULL )` over `if( hs == NULL )` where `hs = ssl->handshake`?

Reply:
It's really not important, but I think that this test checks a property of the `ssl` object rather than the helper `hs` variable, which is why I would personally write it that way, and only use `hs` as a valid and usable pointer to the handshake params object.